configure.ac

```diff
@@ -526,6 +526,73 @@ if test "x$enable_unwind" = xno; then
    AC_SEARCH_LIBS([backtrace], [execinfo])
 fi
 
+AC_ARG_ENABLE([backtrace],
+              [AS_HELP_STRING([--enable-backtrace=<unwind-ptrace>],
+              [enable printing backtraces.])],
+              [],
+              [enable_backtrace=no])
+case "$enable_backtrace" in
```

Comment:
@MrCirdo 
Code simplification and bug fix:
* Since you are going to link to both libunwind-ptrace and libunwind-generic, it is unnecessary to use two AC_CHECK_LIB macros and the way you wrote it won't work.
* `AC_CHECK_HEADERS([libunwind-ptrace.h], ...)` and `AC_CHECK_HEADERS([libunwind/libunwind-ptrace.h], ...)` would define different macros. It doesn't work the way you think. I correct it so that it works the way you intended.

```shell
if test "x$enable_backtrace" = xyes; then
   case $my_htop_platform in
      linux)
         enable_backtrace=unwind-ptrace
         ;;
      *)
         AC_MSG_ERROR([backtrace feature not implemented for the '$my_htop_platform' platform])
         ;;
   esac
fi

case "$enable_backtrace" in
   no)
      ;;
   unwind-ptrace)
      AC_CHECK_LIB([unwind-ptrace], [_UPT_create], [], [AC_MSG_ERROR([can not find required library libunwind-ptrace and libunwind-generic])], [-lunwind-generic])
      AC_CHECK_HEADERS([libunwind-ptrace.h], [], [
         CPPFLAGS="$CPPFLAGS -I/usr/include/libunwind"
         AC_CHECK_HEADERS([libunwind-ptrace.h], [], [AC_MSG_ERROR([can not find required header file libunwind-ptrace.h])])
      ])
      ;;
   *)
      AC_MSG_ERROR([bad value '$enable_backtrace' for --enable-backtrace])
      ;;
esac
if test "x$enable_backtrace" != xno; then
   AC_DEFINE([HAVE_BACKTRACE], [1], [Define if the backtrace feature is enabled])
fi
```

Reply:
Oh okay, it's much clearer than what I wrote 😄 
For some obscure linking error, two `AC_CHECK_LIB` are needed.
Thank you very much @Explorer09